diff --git a/pythonforandroid/build.py b/pythonforandroid/build.py
index d88fdb1f..f430fd20 100644
--- a/pythonforandroid/build.py
+++ b/pythonforandroid/build.py
@@ -875,7 +875,7 @@ def run_pymodules_install(ctx, arch, modules, project_dir=None,
         base_env["PYTHONPATH"] = ctx.get_site_packages_dir(arch)
         info('Upgrade pip to latest version')
         shprint(sh.bash, '-c', (
-            "source venv/bin/activate && pip install -U pip"
+            "source venv/bin/activate && pip install -U 'pip<26'"
         ), _env=copy.copy(base_env))
 
         # Install Cython in case modules need it to build:
@@ -926,7 +926,7 @@ def run_pymodules_install(ctx, arch, modules, project_dir=None,
 
             shprint(sh.bash, '-c', (
                 "venv/bin/pip " +
-                "install -v --target '{0}' --no-deps -r requirements.txt"
+                "install -v --target '{0}' --platform android_24_arm64_v8a --python-version 3.13 --implementation cp --only-binary=:all: --no-deps -r requirements.txt"
             ).format(ctx.get_site_packages_dir(arch).replace("'", "'\"'\"'")),
                     _env=copy.copy(env))
 
